src/NuGet.Clients/NuGet.VisualStudio.Common/ErrorListTableDataSource.cs

```diff
@@ -168,17 +169,30 @@ public void AddNuGetEntries(params ErrorListTableEntry[] entries)
         /// <summary>
         /// Show error window.
         /// </summary>
-        public void BringToFront()
+        public void BringToFrontIfSettingsPermit()
         {
             EnsureInitialized();
 
             NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
             {
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                // Give the error list focus.
-                var vsErrorList = _errorList as IVsErrorList;
-                vsErrorList?.BringToFront();
+                IVsShell vsShell = await _asyncServiceProvider.GetServiceAsync<IVsShell>();
+                object propertyShowTaskListOnBuildEnd;
+                int getPropertyReturnCode = vsShell.GetProperty((int)__VSSPROPID.VSSPROPID_ShowTasklistOnBuildEnd, out propertyShowTaskListOnBuildEnd);
+                bool showErrorListOnBuildEnd = true; //Default to showing the Error List.
+
+                if (getPropertyReturnCode == VSConstants.S_OK)
+                {
+                    bool.TryParse(propertyShowTaskListOnBuildEnd?.ToString(), out showErrorListOnBuildEnd);
```

Comment:
There's a flaw here and it stems from ignoring the return result of `bool.TryParse(…)`.

See if you can find the bug:
```C#
bool a = true;
bool b = bool.TryParse("c", out a);

Console.WriteLine($"a == {a}");
Console.WriteLine($"b == {b}");
```

Reply:
I did not realize that if TryParse fails, it sets the out param regardless. Good find. It hopefully shouldn't happen that a non-bool property is returned from VS, but thought best to guard just in case.